File: fixtures/uri.py
#! /usr/bin/python
from rfc3986 import urlparse, uri_reference
#uri = urlparse('oasis:names:specification:docbook:dtd:xml:4.1.2')
#uri = urlparse('http://httpbin.org/get?utf8=\xe2\x98\x83')
#uri = urlparse('mailto:user@domain.com')
uri = urlparse('mailto://user@domain.com/abc#a?b')
print uri
uri = uri_reference('//abc#a?b')
print uri
uri = uri_reference('')
print uri
uri = uri_reference('../dir')
print uri
uri = uri_reference('file.html')
print uri

File: fixtures/uri.pl
#! /usr/bin/perl
use strict;
use Data::Validate::URI qw(is_uri);
use Data::Dumper;

#if (is_uri('http://c:/directory/file')) {
if (is_uri('http://httpbin.org/get?utf8=\xe2\x98\x83')) {
    print "ok\n";
} else {
    print "error\n";
};
